staticcheck/util/util.go

```diff
@@ -32,8 +32,27 @@ var Analyzers = func() map[string]*analysis.Analyzer {
 
 func FindAnalyzerByName(name string) *analysis.Analyzer {
 	if a, ok := Analyzers[name]; ok {
-		return a
+		return wrapAnalyzer(a)
 	}
 
 	panic(fmt.Sprintf("not a valid staticcheck analyzer: %s", name))
 }
+
+func wrapAnalyzer(a *analysis.Analyzer) *analysis.Analyzer {
```

Comment:
Nit: add a small doc here for easier quick reading. It would be nicer to rename the func to `wrapWithIgnore` or something that immediately communicates to future readers its purpose.

```suggestion
// wrapAnalyzer modifies the original staticcheck's analyzer report and filters out
// issues on lines or files that was marked as ignored by staticcheck's directive.
func wrapAnalyzer(a *analysis.Analyzer) *analysis.Analyzer {
```